### ISSUE TITLE
NPE in Mapsforge PositionLayer.draw

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#14295]
Reported by Play Store for current release 2023.05.15:

```
Exception java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
  at org.mapsforge.map.android.graphics.AndroidBitmap.getWidth (AndroidBitmap.java:114)
  at cgeo.geocaching.maps.mapsforge.v6.layers.PositionLayer.draw (PositionLayer.java:84)
  at org.mapsforge.map.layer.LayerManager.doWork (LayerManager.java:95)
  at org.mapsforge.map.util.PausableThread.run (PausableThread.java:149)
```

**Comment by eddiemuc** [comment]
> I don't get this one. There's a null check 3 lines above the NPE?!

`localArrow` is actually an instance of class `org.mapsforge.map.android.graphics.AndroidBitmap` (as can be seen from the stack trace) which overrides class Bitmap. In this classes' getWidth method the NPE happens. Obviously the object is already destroyed (making the surrounded actual bitmap null).

**Comment by moving-bits** [comment]
Good catch, thanks! Checking the embedded bitmap in an `AndroidBitmap` against `null` can be done by checking for `isDestroyed()`. PR #14312 adds this check.

**Comment by eddiemuc** [comment]
The question here is rather: why is the bitmap destroyed when we use it? I fear there's some higher level logic error in our code.

**Comment by moving-bits** [comment]
Won't fix. Old map implementations are deprecated as of release 2025.07.17 and superseded by UnifiedMap.
